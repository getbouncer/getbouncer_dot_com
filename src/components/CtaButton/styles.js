import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { theme } from '~styles'

export const Button = styled.button`
  background-color: ${props => (props.inverted ? '#fff' : theme.colors.blue)};
  border: 0;
  border-radius: 20px;
  color: ${props => (props.inverted ? theme.colors.blue : '#fff')};
  cursor: pointer;
  padding: 8px 18px;
`

export const ButtonLink = styled.a`
  background-color: ${props => (props.inverted ? '#fff' : theme.colors.blue)};
  border-radius: 20px;
  color: ${props => (props.inverted ? theme.colors.blue : '#fff')};
  padding: 8px 18px;
`

export const InternalLink = styled(Link)`
  background-color: ${props => (props.inverted ? '#fff' : theme.colors.blue)};
  border-radius: 20px;
  color: ${props => (props.inverted ? theme.colors.blue : '#fff')};
  padding: 8px 18px;
`
