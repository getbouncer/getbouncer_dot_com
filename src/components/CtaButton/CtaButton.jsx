import React from 'react'
import PropTypes from 'prop-types'
import { Button, ButtonLink, InternalLink } from './styles'

const CtaButton = ({ inverted, isInternal, text, to }) => {
  if (to) {
    if (isInternal) {
      return (
        <InternalLink to={to} inverted={inverted}>
          {text}
        </InternalLink>
      )
    }

    return (
      <ButtonLink href={to} inverted={inverted}>
        {text}
      </ButtonLink>
    )
  }

  return <Button inverted={inverted}>{text}</Button>
}

CtaButton.propTypes = {
  inverted: PropTypes.bool,
  isInternal: PropTypes.bool,
  text: PropTypes.string.isRequired,
  to: PropTypes.string,
}

CtaButton.defaultProps = {
  inverted: false,
  isInternal: false,
  to: null,
}

export default CtaButton
