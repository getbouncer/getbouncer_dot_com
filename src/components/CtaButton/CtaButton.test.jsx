import React from 'react'
import renderer from 'react-test-renderer'
import CtaButton from './CtaButton'

/* eslint-disable no-undef */

describe('CtaButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CtaButton text="Click Me" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('CtaButton Internal', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CtaButton isInternal text="Click Me" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('CtaButton Inverted', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CtaButton inverted text="Click Me" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
