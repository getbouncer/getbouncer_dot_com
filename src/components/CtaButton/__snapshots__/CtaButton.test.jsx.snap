// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButton Internal renders correctly 1`] = `
<button
  className="css-12unjc5"
>
  Click Me
</button>
`;

exports[`CtaButton Inverted renders correctly 1`] = `
<button
  className="css-1stx2ln"
>
  Click Me
</button>
`;

exports[`CtaButton renders correctly 1`] = `
<button
  className="css-12unjc5"
>
  Click Me
</button>
`;
